playbooks/common/openshift-cluster/redeploy-certificates.yml

```diff
@@ -0,0 +1,210 @@
+    when: openshift_master_ha | bool and openshift_master_cluster_method == 'native'
+  - name: stop master controllers
+    service: name={{ openshift.common.service_type }}-master-controllers state=stopped
+    when: openshift_master_ha | bool and openshift_master_cluster_method == 'native'
+
+- name: Start master services
+  hosts: oo_masters_to_config
+  serial: 1
+  vars:
+    openshift_master_ha: "{{ groups.oo_masters_to_config | length > 1 }}"
+  tasks:
+  - name: start master
+    service: name={{ openshift.common.service_type }}-master state=started
+    when: not openshift_master_ha | bool
+  - name: start master api
+    service: name={{ openshift.common.service_type }}-master-api state=started
+    when: openshift_master_ha | bool and openshift_master_cluster_method == 'native'
+  - name: start master controllers
+    service: name={{ openshift.common.service_type }}-master-controllers state=started
+    when: openshift_master_ha | bool and openshift_master_cluster_method == 'native'
+
+- name: Restart masters (pacemaker)
+  hosts: oo_first_master
+  vars:
+    openshift_master_ha: "{{ groups.oo_masters_to_config | length > 1 }}"
+  tasks:
+  - name: restart master
+    command: pcs resource restart master
+    when: openshift_master_ha | bool and openshift_master_cluster_method == 'pacemaker'
+
+- name: Restart nodes
+  hosts: oo_nodes_to_config
+  tasks:
+  - name: restart node
+    service: name={{ openshift.common.service_type }}-node state=restarted
+
+- name: Redeploy OpenShift Registry
+  hosts: oo_first_master
+  tasks:
+  - name: Create temp directory for kubeconfig
+    command: mktemp -d /tmp/openshift-ansible-XXXXXX
+    register: mktemp
+    changed_when: False
+  - name: Copy the admin client config(s)
+    command: >
+      cp {{ openshift.common.config_base }}/master//admin.kubeconfig {{ mktemp.stdout }}/admin.kubeconfig
+    changed_when: False
+  - name: Determine if registry deployed
+    command: "oc --config={{ mktemp.stdout }}/admin.kubeconfig get dc/docker-registry -o template --template='{{ '{{' }}range .spec.template.spec.containers{{ '}}' }}{{ '{{' }}.livenessProbe.httpGet.scheme{{ '}}' }}{{ '{{' }}end{{ '}}' }}' -n default"
+    register: docker_registry
+    changed_when: false
+    failed_when: false
+  - name: Update registry CA
+    shell: >
+      oc env dc/docker-registry --overwrite OPENSHIFT_CA_DATA="$(cat {{ openshift.common.config_base }}/master/ca.crt)"
+    when: docker_registry.stdout == 'HTTP'
+  - name: Update registry cert
+    shell: >
+      oc env dc/docker-registry --overwrite OPENSHIFT_CERT_DATA="$(cat {{ openshift.common.config_base }}/master/openshift-registry.crt)"
+    when: docker_registry.stdout == 'HTTP'
+  - name: Update registry key
+    shell: >
+      oc env dc/docker-registry --overwrite OPENSHIFT_KEY_DATA="$(cat {{ openshift.common.config_base }}/master/openshift-registry.key)"
+    when: docker_registry.stdout == 'HTTP'
```

Comment:
The default certificates will be part of the HTTP based registry. With HTTPS, custom certificates could be configured and replacing them would be the admin's responsibility as this is written. We could check if the old default registry certificate files match what is configured for the HTTPS registry but I'm not sure how often that will be the case.